File: designPatterns/Behavioral_patterns/Observer/javascript/ConcreteObserverA.js
const Observer = require("./Observer");
const ConcreteSubject = require("./ConcreteSubject");

/**
 * Concrete Observers react to the updates issued by the Subject they had been
 * attached to.
 */
class ConcreteObserverA extends Observer {
  update(subject) {
    if (subject instanceof ConcreteSubject && subject.state < 3) {
      console.log("ConcreteObserverA: Reacted to the event.");
    }
  }
}

module.exports = ConcreteObserverA;

File: designPatterns/Behavioral_patterns/Observer/python/ConcreteObserverA.py
from Observer import Observer


class ConcreteObserverA(Observer):
    """
    Concrete Observers react to the updates issued by the Subject they had been
    attached to.
    """

    def update(self, subject):
        if subject._state < 3:
            print("ConcreteObserverA: Reacted to the event")
